py/src/pathfinder_worker/call.py

```diff
@@ -879,12 +872,48 @@ def create_general_config(chain_id):
     )
 
     assert general_config.cairo_resource_fee_weights[f"{N_STEPS_RESOURCE}"] == 0.05
-    assert (
-        general_config.cairo_resource_fee_weights[f"{BITWISE_BUILTIN}_builtin"] == 12.8
-    )
 
     return general_config
 
 
+def get_resource_fee_weights(starknet_version: Optional[str]):
+    if starknet_version is None:
+        return resource_fee_weights_pre_0_10_2
+
+    try:
+        version = semver.VersionInfo.parse(starknet_version)
+    except ValueError:
+        # we just handle invalid version numbers as old versions
+        return resource_fee_weights_pre_0_10_2
```

Comment:
Should we not make this the latest values instead? I'm thinking if they do something weird in the future, the fee's suddenly teleport backwards in time.

And I think being wrong for `latest` or `pending` is worse than being wrong for historical blocks.

Reply:
We could -- the whole thing is just a bit theoretical though. I'd expect that we _never_ get invalid (non-semver) version numbers. We could also just not catch this exception so that calls would just fail.

Reply:
The version comes from StarkWare (or whoever runs a mock gateway if we allow custom ones) so it should in theory be treated as dirty / suspect.

Reply:
I was wondering if just letting it fail is a better option.

Reply:
Yeah, but then you have invalid data in your database. I don't think there's anything wrong with `starknet_estimateFee` returning an error instead of just masking that you have invalid data and returning _something_?

Reply:
There is no specification saying that this version must be semver. And if you look at our code for this on the rust side, I suspect you'll see Joonas still puts the string into the database if its new, regardless of the semverness of it.

--edit-- and ideed it is just treated as a string specifically for this reason currently.

We can make an argument that `pathfinder` should ensure this is semver and only accept those blocks. But it has not done this to date.

Reply:
I'm not going to block on this issue, so will approve and whichever course you think makes sense is fine 👍